File: archive/iteration_1/tests/regression/sources/data/symbol_sample.py
"""Sample Python file for symbol extraction regression tests."""


class Calculator:
    """A simple calculator class."""

    def __init__(self, value=0):
        self.value = value

    def add(self, x):
        self.value += x
        return self

    def result(self):
        return self.value


def standalone_helper(items):
    """A standalone function."""
    return [item * 2 for item in items]

File: archive/iteration_1/tests/regression/sources/data/symbol_sample.js
/**
 * Sample JavaScript file for symbol extraction regression tests.
 */

function greet(name) {
    return "Hello, " + name + "!";
}

export class Counter {
    constructor(initial = 0) {
        this.count = initial;
    }

    increment() {
        this.count++;
    }

    getCount() {
        return this.count;
    }
}

const double = (x) => {
    return x * 2;
};
